### PR TITLE
V2.18.0 update

### PR DESCRIPTION
**Pull request by BaoTon**
Update to 2.18.0.

All issues stem from reality data and classifiers.

### CODE REVIEW

File: src/@itwinjs-sandbox/view/ViewSetup.ts [hunk 1]
```diff
@@ -89,7 +89,7 @@ export class ViewSetup {
         const modelIds = await ViewSetup.getModelIds(imodel);
         const subCategoryIds = await this.getSubCategoryIds(imodel, "S-SLAB-CONC");
         displayStyle.changeBackgroundMapProps({
-          planarClipMask: PlanarClipMaskSettings.create(PlanarClipMaskMode.IncludeSubCategories, modelIds, subCategoryIds),
+          planarClipMask: PlanarClipMaskSettings.createForElementsOrSubCategories(PlanarClipMaskMode.IncludeSubCategories, subCategoryIds, modelIds).toJSON(),
```
Comment (williamkbentley) [inline src/@itwinjs-sandbox/view/ViewSetup.ts:92]: Apparently you cannot leave a file level comment, or a comment on an unedited line... Unless I missed something.... But What is .ViewSetup.tsx.swp used for?

File: src/frontend-samples/reality-data-sample/RealityDataApi.ts [hunk 2]
```diff
@@ -48,13 +48,15 @@ export default class RealityDataApi {
   // START TRANSPARENCY
   // Modify reality data background transparency using the Viewport API
   public static async setRealityDataTransparency(vp: ScreenViewport, transparency: number) {
-    // For this example we want to affect the appearance of *all* reality models. Therefore, we use -1 as the index.
+    // For this example we want to affect the appearance of *all* reality models. huh?
```
Comment (williamkbentley) [inline src/frontend-samples/reality-data-sample/RealityDataApi.ts:51]:
```suggestion
    // For this example we want to affect the appearance of *all* reality models.
```

Comment (williamkbentley) [reply src/frontend-samples/reality-data-sample/RealityDataApi.ts:51]: The README is also now out of date

**Review by williamkbentley — CHANGES_REQUESTED** [review]
Few comments

File: src/test/frontend-sample-showcase.test.tsx [hunk 3]
```diff
@@ -197,20 +201,26 @@ describe("Classifers", () => {
...
+        flags
+      );
+
       ClassifierApi.updateRealityDataClassifiers(vp, testClassifier);
-      const style = vp.displayStyle.clone();
-      style.forEachRealityModel(
-        (model) => {
-          expect(model.classifiers?.length).to.equal(1);
+      vp.displayStyle.settings.contextRealityModels.models.forEach((model) => {
+        if (model.classifiers) {
```
Comment (williamkbentley) [inline src/test/frontend-sample-showcase.test.tsx:217]: Wouldn't the model not having a classifier indicate a test failure?

File: src/@itwinjs-sandbox/view/ViewSetup.ts (hunk 1 above)
Comment (BaoTon) [reply src/@itwinjs-sandbox/view/ViewSetup.ts:92]: Removed stray .swp file - this is used for vim which is my file editor.

File: src/test/frontend-sample-showcase.test.tsx (hunk 3 above)
Comment (BaoTon) [reply src/test/frontend-sample-showcase.test.tsx:217]: The test fails 100% of the time during test-certa, wasn't there something special you had to do to add the classifier attribute?

File: src/frontend-samples/reality-data-sample/RealityDataApi.ts (hunk 2 above)
Comment (BaoTon) [reply src/frontend-samples/reality-data-sample/RealityDataApi.ts]: Updated

File: src/test/frontend-sample-showcase.test.tsx (hunk 3 above)
Comment (williamkbentley) [reply src/test/frontend-sample-showcase.test.tsx:217]: fixed

**Review by williamkbentley — APPROVED** [review]

**Pull request merged**